lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1590,6 +1590,33 @@ class HSAQueue final : public KalmarQueue
     };
 
 
+    void drain() {
+      while (1) {
```

Comment:
Also if it's possible to forcibly abort `drain()` by letting the function check an flag?
I can foresee we may possible run into soft hang here in case the hardware goes wrong.

Reply:
who will set the flag and how it will be set?

Reply:
how about let the flag be an environment variable? users / sysadmins could set such an environment variable to break this loop and let the application abort, shall the application acts funky.

It could be an overkill though. I am just not used to see a `while(1)` loop in runtime without an explicit way to deterministically escape from the loop.

Reply:
updated, the exit condition is not controlled by whether a valid is found